agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -618,3 +618,197 @@ func TestNetworkModeBridge(t *testing.T) {
 	err := networkModeTest(t, agent, "bridge")
 	require.NoError(t, err, "Networking mode bridge testing failed")
 }
+
+// TestCustomAttributesWithValidOptions tests the ECS_INSTANCE_ATTRIBUTES
+// upon agent registration with a key, value pair
+func TestCustomAttributesWithValidOptions(t *testing.T) {
+	len := 25
+	key, value := GetRandomString(len), GetRandomString(len)
+
+	customAttribute := fmt.Sprintf("{\"%s\": \"%s\"}", key, value)
```

Comment:
@petderek In this case, the random string comes from a restricted and controlled set of characters which stick to the specs imposed. I was under the wrong impression that it makes sense for the functional tests to verify the boundary conditions under a genuine use-case. However, I've now removed all the edge case testing from here.